osidb/models.py

```diff
@@ -2960,7 +2960,16 @@ def create_package(flaw, package, **extra_fields):
 
 
 class Package(ACLMixin, BugzillaSyncMixin, TrackingMixin):
-    """Model representing a package with connected versions."""
+    """
+    Model representing a package with connected versions.
+
+    The model's structure allows future extensibility with features from the CVE 5.0 schema.
+    Currently, it tracks only versions corresponding to bugzilla fixed_in field, which correspond to
+    CVE 5.0's `"status": "unaffected"`.
+    References:
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/docs/versions.md
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
```

Comment:
Let's not base the model on something that we don't use, adhere to fully or can otherwise be arranged at the representation layer

Reply:
I agree with Adrian, and I would prefer to discuss it on the team call. Even if we seem to be compatible with the new schema now, something may happen in the future that will force new changes/redesign again.

Reply:
Would it still be compatible with the existing API or would it lead to a breaking changes? If the latter we should communicate it with the SDEngine guys. The naming change was done by @jasinner so he could comment on the reasoning and which one makes more sense.

Reply:
I guess we can do anything we want and just implement custom logic for the serializer behind [Flaw's `package_versions` read-only field (hyperlink to current master)](https://github.com/RedHatProductSecurity/osidb/blob/9533bb350265a73ca5d7c6bb5e5e70293ac91969/osidb/serializer.py#L808) to maintain compatibility with **existing** API and existing users.

As for the **_new_** API _added_ in this PR, that's not used by anyone else yet (obviously), so there (`/api/v1/flaws/CVE-xxx-xxx/package_versions`) we can do whatever we want and name it whatever we want (perhaps `/api/v1/flaws/CVE-xxx-xxx/fixed_in`?).

As to what's feasible and not overly complex to design and maintain, I propose choosing one of these two paths:
- Maintain CVE 5.0 compatibility and maintain current model structure, so that we can use standard serializer mixins, standard schema autogeneration, standard viewset and as small serializer changes as possible.
- Drop CVE 5.0 compatibility, make it into anything that makes sense, and have the default the API structure that results from the standard model structure (as used in other osidb models), standard osidb mixins, standard viewsets, standard auto-generated schema, etc, just as with any other OSIDB model.

I specifically advise to avoid trying to have one seemingly clear model structure and another seemingly clear API structure, because in between would be bespoke unmaintainable glue.

Reply:
I agree. One or another. The first one is I think what SDEngine currently expects. What is the advantage of the second one? Simpler model I guess. That would however require SDEngine functionality rewrite, correct?

Reply:
> would however require SDEngine functionality rewrite

I don't think so. SDEngine uses only whatever was there before this PR. Which is the read-only field directly on Flaw. And this is the only serializer where manually creating an arbitrary data structure is without negative effects on maintainability. So we can go with the second option and still have this field return the structure expected by SDEngine.

Reply:
Closed as per https://github.com/RedHatProductSecurity/osidb/pull/307#issuecomment-1713798433